### ISSUE TITLE
Why a "%d" specifier next to another "%d" specifier would crash the decompressor 

### ISSUE THREAD

**Issue by zcq1223** [PlatformLab/NanoLog#41]
Make modifications  to "sample/main.cc" as shown in the figure below，
![image](https://user-images.githubusercontent.com/34498523/96364657-3fd60f00-116e-11eb-9880-1f022be98dee.png)
Then compile and run the program,
```bash
$ cd sample
$ make
$ ./sampleApplication
$ ./decompressor decompress compressedLog
```
![image](https://user-images.githubusercontent.com/34498523/96364784-e28e8d80-116e-11eb-81c1-50df6a175417.png)
What I expected to output is "My number is 987654321123456789", but the actual output is "My number is 987654321987654321".  It outputs the integer 987654321 twice but doesn't output the integer 123456789. 
On top of that, the decompressor aborted.

adding a comma between the two "%d" specifiers could solve the problem,
![image](https://user-images.githubusercontent.com/34498523/96365563-17511380-1174-11eb-9819-5c23aa7e7635.png)

I'm confusing and wondering whether this is a bug or not.

**Comment by syang0** [comment]
Hey zcq1223,

Thanks for the clear examples! I was able to reproduce the issue, and I do believe it is in fact a bug; NanoLog should support logging `%d%d` as two integers.

I briefly looked at the stack trace, and it appears to be crashing at [Log.cc:1405](https://github.com/PlatformLab/NanoLog/blob/64020673436e37ec5198854bedc7aac98419e323/runtime/Log.cc#L1406), and I suspect it's due to an alignment issue when reading back the integer arguments.

In any case, I should be able to track down the bug and fix it in a couple of days, so please use the comma alternative for now. Thanks for reporting!

Best Regards,
Stephen Yang

**Comment by syang0** [comment]
Hey zcq1223,

The issue should be resolved now with the latest commit (b1b9ee0). Please check it out and let me know if the issue persists.

Best Regards,
Stephen Yang


P.S. If you're curious, the fix just involved adding one line to [Log.cc](https://github.com/PlatformLab/NanoLog/blob/b1b9ee0876969576470231c38ecdd099a3aeb952/runtime/Log.cc#L890); the rest of the commit adds unit tests.

**Comment by zcq1223** [comment]
Great, I checked it out and found the issue was resolved.
